src/wrappers/themis/java/com/cossacklabs/themis/SecureSession.java

```diff
@@ -116,7 +116,7 @@ public byte[] getData() {
 	native long create(byte[] id, byte[] signKey);
 	native void destroy();
 	
-	native byte[] jniGenerateConntect();
+	native byte[] jniGenerateConnect();
```

Comment:
I support fixing the function name. However, this needs a bit more work.

---

First, the function name needs to be updated in C code as well:

https://github.com/cossacklabs/themis/blob/5d4a0b707c33b32d7a1f2afc0805fb7b1fa49b8d/jni/themis_session.c#L469-L470

```diff
-Java_com_cossacklabs_themis_SecureSession_jniGenerateConntect
+Java_com_cossacklabs_themis_SecureSession_jniGenerateConnect
```

(mind the code format, it will require parameter list to be neatly aligned)

---

Second, even though JNI interface is a private one, I'd keep the old symbol name around to avoid breaking applications that were built using the old JavaThemis. When they upgrade the JNI library, they should keep working without having to recompile their Java code using the new version of JavaThemis. 

```c
JNIEXPORT jbyteArray JNICALL Java_com_cossacklabs_themis_SecureSession_jniGenerateConntect(JNIEnv* env, 
                                                                                           jobject thiz)
__attribute__((alias("Java_com_cossacklabs_themis_SecureSession_jniGenerateConnect")));
```

With this added, users of the new version will use the "good" name, and any new code should be written using this name. However, the function will also be available under the old name for binaries that were linked against that name.